vue-components/stories/Icon.stories.ts

```diff
@@ -8,7 +8,9 @@ export default {
 };
 
 const typeValues = Object.values( IconTypes );
-export function basic( args: object ): Component {
+export function basic( args: { dir: string; type: string }, { argTypes } ): Component {
+	argTypes.dir.control.disable = !flippable.includes( args.type );
```

Comment:
I think this might need to exist in the component's `data` or as a computed property in order for it to be responsive to changes in the canvas view. I had a similar problem in TextArea. See https://github.com/wmde/wikit/blob/master/vue-components/stories/TextArea.stories.ts#L11-L18

Reply:
Thanks for the comment! Sadly I'm not sure I understand the suggestion. The component reacts fine to any changes coming from the controls. It's the other way around that seems to get lost somehow. Setting `argTypes.dir.control.disable` doesn't reach the controls. The line we're commenting on gets executed at the right time with the right values, it just doesn't have the same effect as it does in the "Docs" tab.

In the "Canvas" tab, the component sits on a separate page that's embedded through an iframe, in "Docs" they're on the same page. I sadly don't know storybook well enough to know if and how it's possible to make the two talk both ways in the former setup.

Feel free to tinker with this and build on top or try a different approach. I probably won't have time to look into this again this week, but would of course be happy to see it done :smile: 

Ideally this wouldn't require a control -> component story -> control roundtrip at all. This would be easy if one control could inform another. That's https://github.com/storybookjs/storybook/issues/11486 I guess.

Reply:
Yeah, I guess I misunderstood how this works. Now that I tried it, I see it's not what I thought :facepalm:  

In any case, it might be too much of an overhead to try and shoehorn an unsupported feature into storybook for a component that is probably going to be migrated to codex (and with that into vitepress) probably sooner rather than later.

Reply:
Ok. Considering Storybook's limitations and the effort required to only potentially overcome them, I'd say we should close this PR after all? Displaying the desired behavior only in Docs might make the improvement less discoverable, and the inconsistency between both Controls panels could generate confusion.
Thanks, Jakob, for getting so close to a solution! 💯 And thanks Itamar for your support 🙌

Reply:
It has all the thumbs up, and I absolutely agree that this isn't worth the effort and/or potential for confusion. Thanks for looking into it though! :sparkles: